application/ledger_query_service/src/ethereum/queries.rs

```diff
@@ -406,24 +381,6 @@ mod tests {
         assert_that!(query.matches_transaction_receipt(receipt)).is_false()
     }
 
-    #[test]
-    fn given_a_transaction_receipt_should_not_match_empty_query() {
-        let redeem_log_msg =
-            "0xB8CAC300E37F03AD332E581DEA21B2F0B84EAAADC184A295FEF71E81F44A7413".into();
-
-        let query1 = LogQuery {
-            topics: vec![vec![]],
-        };
-        let log1 = log(vec![redeem_log_msg]);
-        let receipt1 = transaction_receipt(vec![log1]);
-        assert_that!(query1.matches_transaction_receipt(receipt1)).is_false();
-
-        let query2 = LogQuery { topics: vec![] };
-        let log2 = log(vec![redeem_log_msg]);
-        let receipt2 = transaction_receipt(vec![log2]);
-        assert_that!(query2.matches_transaction_receipt(receipt2)).is_false()
```

Comment:
❌Why did you remove this test? Shouldn't this still apply, even if you remove the empty check?

Reply:
Maybe I'm understanding it wrong, but if we remove the empty check then empty queries are allowed and they would match everything, so this test would not pass.

Reply:
ah, wasn't aware that this is what we want